tests/conftest.py

```diff
@@ -19,10 +22,137 @@
+    variables:
+
+        - REDSHIFT_HOST
+        - REDSHIFT_PORT
+        - REDSHIFT_USERNAME
+        - REDSHIFT_DATABASE
+        - PGPASSWORD
+
+    If these conditions are met, the tests will be ran
+    against a real Redshift instance. Otherwise, they
+    will be ran against a mock engine.
+
+    See the fixture, _redshift_database_tool, for usage.
+    """
+    pgpassword = os.environ.get('PGPASSWORD', None)
+    if not pgpassword:
+        pytest.skip('This test will only work on Travis.')
+
+    return {
+        "host": os.getenv("REDSHIFT_HOST", None),
+        "port": os.getenv("REDSHIFT_PORT", None),
+        "username": os.getenv("REDSHIFT_USERNAME", "travis"),
+        "password": pgpassword,
+        "database": os.getenv("REDSHIFT_DATABASE", "dev"),
+        "dialect": redshift_dialect_flavor,
+    }
+
+
+@pytest.fixture(scope="session")
+def aws_account_id():
+    """ The aws_account_id fixture allows the developer to pass in their own
+    AWS_ACCOUNT_ID for other Redshift instances by setting the following
+    environment variable: AWS_ACCOUNT_ID
+    """
+    return os.getenv("AWS_ACCOUNT_ID", '000123456789')
+
+
+@pytest.fixture(scope="session")
+def iam_role_name():
+    """ The iam_role_name fixture allows the developer to pass in their own
+    IAM_ROLE_NAME for other Redshift instances by setting the following
+    environment variable: IAM_ROLE_NAME
+    """
+    return os.getenv("REDSHIFT_IAM_ROLE_NAME", 'redshiftrole')
+
+
+@pytest.fixture(scope="session")
+def aws_partition():
+    """ The aws_partition fixture allows the developer to pass in their own
+    aws partition value for other Redshift instances by setting the following
+    environment variable: AWS_PARTITION
+    """
+    return os.getenv("AWS_PARTITION", 'aws-us-gov')
+
+
+@pytest.fixture(scope="session")
+def iam_role_arn(aws_account_id, iam_role_name):
+    """ The iam_role_arn fixture constructs the ARN for the IAM role.
+
+    The following fixtures are used to construct the ARN:
+
+    - aws_account_id
+    - iam_role_name
+    """
+    return f'arn:aws:iam::{aws_account_id}:role/{iam_role_name}'
+
+
+@pytest.fixture(scope="session")
+def iam_role_arn_with_aws_partition(
+    aws_account_id,
+    aws_partition,
+    iam_role_name
+):
+    """ The iam_role_arn_with_aws_partition fixture allows the developer to
+    pass in their own IAM_ROLE_ARN for other Redshift instances by setting
+    the following environment variable:
+    REDSHIFT_IAM_ROLE_ARN_WITH_AWS_PARTITION
+
+    The following fixtures are used to construct the rest of the ARN:
+
+    - aws_account_id
+    - aws_partition
+    - iam_role_name
+    """
+    return os.getenv(
+        "REDSHIFT_IAM_ROLE_ARN_WITH_AWS_PARTITION",
+        f'arn:{aws_partition}:iam::{aws_account_id}:role/{iam_role_name}'
+    )
+
+
+@pytest.fixture(scope="session")
+def iam_role_arns():
+    """
+    The iam_role_arns fixture allows the developer to pass in their own
```

Comment:
similar comment to above -- what permissions do the roles have, when do you want to define this env var, are these roles associated with the cluster?